### PR TITLE
fix(shared): Add trim to `mergeClasses`

### PR DESCRIPTION
**Pull request by ericfennis**
fixes: #2498

Adds trim to `mergeClasses` function. Possibly fixes hydration issues in NextJS.
Also added a test and adjusted GH actions to run this change on each package.

### CODE REVIEW

File: packages/shared/src/utils.ts [hunk 1]
```diff
@@ -43,4 +43,5 @@ export const mergeClasses = <ClassType = string | undefined | null>(...classes:
     .filter((className, index, array) => {
       return Boolean(className) && array.indexOf(className) === index;
     })
-    .join(' ');
+    .join(' ')
+    .trim();
```
Comment (jguddas) [inline packages/shared/src/utils.ts:47]: What about doing `return Boolean(className.trim()) && array.indexOf(className) === index;` instead?

That way you don't get something weird if you do `mergeClasses('lucide', ' ', 'lucide-circle')`.

Not sure about the performance impact tho.

Comment (ericfennis) [reply packages/shared/src/utils.ts:47]: Hmm good idea, I will add that one too! Performance is negligible.

**Review by jguddas — APPROVED** [review]

**Comment by benjick** [comment]
@ericfennis this needs `prettier -w`

File: packages/shared/src/utils.ts [hunk 2]
```diff
@@ -41,6 +41,7 @@ export const toPascalCase = <T extends string>(string: T): CamelToPascal<T> => {
 export const mergeClasses = <ClassType = string | undefined | null>(...classes: ClassType[]) =>
   classes
     .filter((className, index, array) => {
-      return Boolean(className) && array.indexOf(className) === index;
+      return Boolean(className) && (className as string).trim() !== '' && array.indexOf(className) === index;
```
Comment (benjick) [inline packages/shared/src/utils.ts:44]:
```suggestion
      return (
        Boolean(className) &&
        (className as string).trim() !== '' &&
        array.indexOf(className) === index
      );
```

to pass prettier check

Comment (ericfennis) [reply packages/shared/src/utils.ts:44]: Thanks this looks more readable!

**Pull request merged**